src/test/java/com/napier/sem/DAO_UnitTests.java

```diff
@@ -33,4 +47,220 @@ public void getWhereCondition_areaNameAppearsInWhereCondition()
         // then
         assertTrue(whereCondition.contains(areaName));
     }
+
+    // test that the whereCondition is null when areaName is null
+    @Test
+    public void getWhereCondition_nullAreaNameReturnsNull()
+    {
+        // given
+        String areaName = null;
+
+        // when
+        String whereCondition = DAO.getWhereCondition(App.WORLD, areaName);
+
+        // then
+        assertNull(whereCondition);
+    }
+
+    // test that the whereCondition is null when areaFilter is null
+    @Test
+    public void getWhereCondition_nullAreaFilterReturnsNull()
+    {
+        // given
+        String areaFilter = null;
+
+        // when
+        String whereCondition = DAO.getWhereCondition(areaFilter, "Earth");
+
+        // then
+        assertNull(whereCondition);
+    }
+
+    /**
+     * Unit tests covering the DAO.allCountriesIn method
+     */
+    // test that countries list is empty when areaFilter and areaName are null
+    @Test
+    public void allCountriesIn_bothArgumentsNullListEmpty()
+    {
+        // given
+        String areaFilter = null;
+        String areaName = null;
+
+        // when
+        ArrayList<Record> countries = dao.allCountriesIn(areaFilter, areaName);
+
+        // then
+        assertEquals(0, countries.size());
+    }
+
+    // test that countries list is empty when areaFilter is null
+    @Test
+    public void allCountriesIn_areaFilterNullListEmpty()
+    {
+        // given
+        String areaFilter = null;
+
+        // when
+        ArrayList<Record> countries = dao.allCountriesIn(areaFilter, "Scotland");
+
+        // then
+        assertEquals(0, countries.size());
+    }
+
+    // test that countries list is empty when areaName is null
+    @Test
+    public void allCountriesIn_areaNameNullListEmpty()
+    {
+        // given
+        String areaName = null;
+
+        // when
+        ArrayList<Record> countries = dao.allCountriesIn(App.COUNTRY, areaName);
+
+        // then
+        assertEquals(0, countries.size());
+    }
+
+    // check test passes with valid areaFilter and areaName
+    @Test
+    public void allCountriesIn_happyPath()
```

Comment:
This is for the tests in general, some set both areaFilter and areaName as strings, some only set one of these.  But looking at it I seems like this is to follow the given when then structure.  If so please disregard.

Reply:
it's to try and make what we're testing more clear. if thye're in the given, we're testing those inputs.

Reply:
Yeah makes sense, that's what I thought after looking at it